File: 0014-longest-common-prefix/0014-longest-common-prefix.c
char* createCopyOf(char* str, size_t len) {
    char* res = (char*)malloc(sizeof(char) * (len + 1));

    if (res == NULL) {
        return NULL;
    }

    res[len] = '\0';
    strcpy(res, str);
    return res;
}

char* longestCommonPrefix(char** strs, int strsSize) {
    if (strs == NULL || strsSize == 0) {
        return "";
    }

    char const* first = strs[0];

    int i = 0;

    while (first[i] != '\0') {
        char fc = first[i];

        for (int j = 1; j < strsSize; ++j) {
            char const* curr = strs[j];
            char cc = curr[i];

            if (cc == '\0' || cc != fc) {
                char* res = (char*)malloc(sizeof(char) * (i + 1));

                if (res == NULL) {
                    return "";
                }

                res[i] = '\0';
                strncpy(res, first, i);
                return res;
            }
        }

        ++i;
    }

    return createCopyOf(first, i);
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if (strs.Length == 0) {
            return "";
        }

        string model = strs[0];
        char[] arr = new char[model.Length];
        int i = 0;

        for (; i < model.Length; ++i) {
            bool allMatching = true;

            for (int j = 1; j < strs.Length; ++j) {
                string curr = strs[j];

                if (i >= curr.Length || curr[i] != model[i]) {
                    allMatching = false;
                    break;
                }
            }

            if (allMatching) {
                arr[i] = model[i];
            } else {
                break;
            }
        }
        
        return new string(arr, 0, i);
    }
}